src/org/labkey/response/ResponseUpgradeCode.java

```diff
@@ -0,0 +1,125 @@
+package org.labkey.response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager;
+import org.labkey.api.data.DeferredUpgrade;
+import org.labkey.api.data.PropertyManager;
+import org.labkey.api.data.UpgradeCode;
+import org.labkey.api.module.Module;
+import org.labkey.api.module.ModuleContext;
+import org.labkey.api.module.ModuleLoader;
+import org.labkey.api.module.ModuleProperty;
+import org.labkey.api.util.StringUtilsLabKey;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Base64.getDecoder;
+import static org.labkey.response.ResponseController.ServerConfigurationAction.RESPONSE_SERVER_CONFIGURATION;
+import static org.labkey.response.ResponseController.ServerConfigurationAction.WCP_SERVER;
+import static org.labkey.response.ResponseModule.METADATA_SERVICE_ACCESS_TOKEN;
+import static org.labkey.response.ResponseModule.METADATA_SERVICE_BASE_URL;
+import static org.labkey.response.ResponseModule.SURVEY_METADATA_DIRECTORY;
+
+public class ResponseUpgradeCode implements UpgradeCode
+{
+    private static final Logger LOG = LogManager.getLogger(ResponseUpgradeCode.class);
+
+    /**
+     * Invoked within mobileappstudy-20.001-20.002.sql
+     * Response Module Properties, which were formerly set in the Folder Management Module Properties tab, are here migrated
+     * to the Admin Console Response Server Configuration page.
+     */
+    @DeferredUpgrade
+    public static void migrateResponseServerConfig(ModuleContext context)
+    {
+        Module responseModule = ModuleLoader.getInstance().getModule("Response");
+        if (responseModule == null)
+            return;
+
+        ModuleProperty responseModulePropertyDirectory = responseModule.getModuleProperties().get(SURVEY_METADATA_DIRECTORY);
+        ModuleProperty responseModulePropertyURL = responseModule.getModuleProperties().get(METADATA_SERVICE_BASE_URL);
+        ModuleProperty responseModuleProperty = responseModule.getModuleProperties().get(METADATA_SERVICE_ACCESS_TOKEN);
```

Comment:
Are you sure this works? You've deleted the module properties and their registration... so I don't see how these gets will return anything but null.

Reply:
Resolved!